gazelle_haskell_modules/rule_generation.go

```diff
@@ -398,17 +398,27 @@ type ModuleImport struct {
 	ModuleName  string
 }
 
+type TempModuleImport struct {
+	IsSourceImported bool
+	ModuleName       []string
+}
+
 func (moduleImport *ModuleImport) UnmarshalJSON(data []byte) error {
-	var aux []string
-	if err := json.Unmarshal(data, &aux); err != nil {
+	var tempMod TempModuleImport
+	if err := json.Unmarshal(data, &tempMod); err != nil {
 		return err
 	}
+	aux := tempMod.ModuleName
 	pkgName := ""
 	if len(aux) > 1 {
 		pkgName = aux[0]
 	}
+	suffix := ""
+	if tempMod.IsSourceImported {
+		suffix = "-boot"
+	}
 	moduleImport.PackageName = pkgName
-	moduleImport.ModuleName = aux[len(aux)-1]
+	moduleImport.ModuleName = aux[len(aux)-1] + suffix
```

Comment:
Why not augment `ModuleImport` with a boolean `IsSourceImported` field instead of putting the information in unstructured fashion as a string suffix?